internal/bminventory/inventory_v2_handlers.go

```diff
@@ -63,7 +63,29 @@ func (b *bareMetalInventory) V2GetCluster(ctx context.Context, params installer.
 }
 
 func (b *bareMetalInventory) V2DeregisterCluster(ctx context.Context, params installer.V2DeregisterClusterParams) middleware.Responder {
-	if err := b.DeregisterClusterInternal(ctx, params); err != nil {
+	log := logutil.FromContext(ctx, b.log)
+	cluster, err := common.GetClusterFromDB(b.db, params.ClusterID, common.UseEagerLoading)
+	if err != nil {
+		return common.NewApiError(http.StatusNotFound, err)
+	}
+
+	if b.ocmClient != nil {
+		if err = b.integrateWithAMSClusterDeregistration(ctx, cluster); err != nil {
+			log.WithError(err).Errorf("Cluster %s failed to integrate with AMS on cluster deregistration", params.ClusterID)
+			return common.NewApiError(http.StatusInternalServerError, err)
+		}
+	}
+
+	if err = b.deleteDNSRecordSets(ctx, *cluster); err != nil {
+		log.Warnf("failed to delete DNS record sets for base domain: %s", cluster.BaseDNSDomain)
+	}
+
+	if err = b.deleteOrUnbindHosts(ctx, cluster); err != nil {
```

Comment:
I don't think we supply dns or AMS integration when using the kubeAPI, right?

My intention was to make the `Internal` function only what is needed for kubeAPI as REST should always be a superset.

Reply:
> Maybe worth a note somewhere that kube-api is using only the Internal func :/

Isn't this the entire purpose of the `Internal*` functions? I said this in the commit message, but I can add some comment to the internal function definition.

```
This PR moves all the SaaS specific logic out of
`DeregisterClusterInternal` and into `V2DeregisterCluster`. This will
ensure that this logic is still run when it is needed, but avoid the
race with the controllers when using the kubeAPI.
```